### DISCUSSION TITLE
Visibility of handle method if used AsObject

### DISCUSSION THREAD

**Discussion by mattsches** [lorisleiva/laravel-actions#179] [Q&A]
Hi everyone, it may be an edge case, but can I safely set the visibility of the `handle` method to `private` if I use my action only `AsObject`?

```php
class GetCustomData
{
    use AsObject;

    private function handle(): array
    {
        $customData = [];
        return $customData;
    }
}
```

**Comment by Wulfheart** [comment]
I don't think so. Because as per the [trait AsObject](https://github.com/lorisleiva/laravel-actions/blob/main/src/Concerns/AsObject.php) it will be executed as this:


```php
public static function make()
{
    return app(static::class);
}

/**
 * @see static::handle()
 * @param mixed ...$arguments
 * @return mixed
 */
public static function run(...$arguments)
{
    return static::make()->handle(...$arguments);
}
``` 

I am not sure how exactly this is going to turn out but as the `run` calls from outside the class, therefore I suppose it needs to be public. However, you may give it a try.
But be aware that the IDE-Helper for this is most likely to not work.

May I ask why you would like to make it private?

**Reply by mattsches** [reply]
Actually, I more or less accidentally set it to private when I was investigating the available traits - and it still worked!

As I understand it, the static `make` method gets an instance of the object from the app container, and because the instance itself uses the trait, the call to `handle` is actually from _inside_ the class.

I assume it's different if the `AsController` trait is used, but I haven't tried it.

So I was thinking of keeping `handle` private, to make it explicit that my class should only be used as an object (as intended), and no one accidentally tries to call `handle` in any other way. Does this make any sense?

**Reply by Wulfheart** [reply]
Oh! This could actually be a nice addition. Pinging @lorisleiva here. What do you think about this concept?
